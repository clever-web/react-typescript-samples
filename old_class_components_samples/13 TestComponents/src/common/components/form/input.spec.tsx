import * as React from 'react';
import { shallow } from 'enzyme';
import { Input } from '.';


describe('/common/components specs',()=>{

    it('should render as expected when passing required properties', () => {
        // Arrange
        const props = {
            name: 'test name',
            label: 'test label',
            value: 'test value',
            onChange: () => { },
        };

        // Act
        const component = shallow(
            <Input
                {...props}
            />,
        );

        // Assert
        expect(component).toMatchSnapshot();
    });
    it('should render as expected when passing required and optional properties', () => {
        // Arrange
        const props = {
            name: 'test name',
            label: 'test label',
            value: 'test value',
            onChange: () => { },
            onBlur: () => { },
            error: 'test error',
            type: 'test type',
        };

        // Act
        const component = shallow(
            <Input
                {...props}
            />,
        );

        // Assert
        expect(component).toMatchSnapshot();
    });
    it('should call to onChange prop when simulate input onChange', () => {
        // Arrange
        const props = {
            name: 'test name',
            label: 'test label',
            value: 'test value',
            onChange: jest.fn(),
            onBlur: () => { },
            error: 'test error',
            type: 'test type',
        };

        // Act
        const component = shallow(
            <Input
                {...props}
            />,
        );

        component.find('input').simulate('change', {
            target: {
                name: 'test name',
                value: 'updated value',
            },
        });

        // Assert
        expect(props.onChange).toHaveBeenCalledWith('test name', 'updated value');
    });

});