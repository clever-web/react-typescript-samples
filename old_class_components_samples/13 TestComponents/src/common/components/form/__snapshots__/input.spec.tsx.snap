// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/common/components specs should render as expected when passing required and optional properties 1`] = `
<div
  className="form-group has-error"
>
  <label
    htmlFor="test name"
  >
    test label
  </label>
  <div
    className="field"
  >
    <input
      className="form-control"
      name="test name"
      onChange={[Function]}
      type="text"
      value="test value"
    />
  </div>
  <div
    className="help-block"
  >
    test error
  </div>
</div>
`;

exports[`/common/components specs should render as expected when passing required properties 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="test name"
  >
    test label
  </label>
  <div
    className="field"
  >
    <input
      className="form-control"
      name="test name"
      onChange={[Function]}
      type="text"
      value="test value"
    />
  </div>
  <div
    className="help-block"
  />
</div>
`;
